Shader improvements

mtrive opened the pull request:
Shaders improvements:
- Add "Num Variants" property to reported shaders

![num-variants2](https://user-images.githubusercontent.com/12098182/192544448-71bb125f-72f5-4283-a9af-1a06123a20c1.PNG)


- Add Compute Shader Variants support

![compute-shaders](https://user-images.githubusercontent.com/12098182/192543386-4d6b1c13-633e-4699-9648-e1fe6fc2e697.PNG)

- This PR also fixes the over-reporting of built shader variants count.

cinight reviewed (approved):
lgtm! Thanks for adding this.

unitystevem reviewed (changes requested):
As discussed, I think ShaderTypeIsFragment() needs a comment explaining the context around why it does what it does with OpenGL/ES shaders.

More seriously, I'm seeing a divide by 0 exception on line 411 of ShadersModule.cs. I guess (numCompilerPlatforms == 0) for some shaders? Error message and partial callstack below. Repro steps are:

- Add this branch of PA to a new project with the LEGO Microgame, 2021.3.7f1
- Open Project Auditor, go to Shader Variants window, click Clear
- Build the game (I build OSX Standalone)
- In the Shader Variants window, click Refresh.

------

DivideByZeroException: Attempted to divide by zero.
Unity.ProjectAuditor.Editor.Modules.ShadersModule.AddShader (UnityEngine.Shader shader, System.String assetPath, System.String assetSize, System.Boolean isAlwaysIncluded, System.Action`1[T] onIssueFound) (at Library/PackageCache/com.unity.project-auditor@c73131f0c1/Editor/Modules/ShadersModule.cs:411)
Unity.ProjectAuditor.Editor.Modules.ShadersModule.ProcessShaders (System.Collections.Generic.Dictionary`2[TKey,TValue] shaderPathMap, System.Action`1[T] onIncomingIssues) (at Library/PackageCache/com.unity.project-auditor@c73131f0c1/Editor/Modules/ShadersModule.cs:368)
Unity.ProjectAuditor.Editor.Modules.ShadersModule.Audit (Unity.ProjectAuditor.Editor.ProjectAuditorParams projectAuditorParams, Unity.ProjectAuditor.Editor.IProgress progress) (at Library/PackageCache/com.unity.project-auditor@c73131f0c1/Editor/Modules/ShadersModule.cs:248)
Unity.ProjectAuditor.Editor.UI.ProjectAuditorWindow.AuditSingleModule (Unity.ProjectAuditor.Editor.Core.ProjectAuditorModule module) (at Library/PackageCache/com.unity.project-auditor@c73131f0c1/Editor/UI/ProjectAuditorWindow.cs:675)
Unity.ProjectAuditor.Editor.UI.ProjectAuditorWindow.AuditSingleModule[T] () (at Library/PackageCache/com.unity.project-auditor@c73131f0c1/Editor/UI/ProjectAuditorWindow.cs:634)
Unity.ProjectAuditor.Editor.UI.ProjectAuditorWindow.AnalyzeShaderVariants () (at Library/PackageCache/com.unity.project-auditor@c73131f0c1/Editor/UI/ProjectAuditorWindow.cs:680)
Unity.ProjectAuditor.Editor.UI.ProjectAuditorWindow+<>c.<OnLoad>b__32_9 () (at Library/PackageCache/com.unity.project-auditor@c73131f0c1/Editor/UI/ProjectAuditorWindow.cs:341)
Unity.ProjectAuditor.Editor.UI.Framework.AnalysisView.DrawToolbarButton (UnityEngine.GUIContent guiContent, System.Action onClick) (at Library/PackageCache/com.unity.project-auditor@c73131f0c1/Editor/UI/Framework/AnalysisView.cs:576)

mtrive commented:
> As discussed, I think ShaderTypeIsFragment() needs a comment explaining the context around why it does what it does with OpenGL/ES shaders. 

Good point. Added a comment.

> More seriously, I'm seeing a divide by 0 exception on line 411 of ShadersModule.cs. I guess (numCompilerPlatforms == 0) for some shaders? Error message and partial callstack below. Repro steps are:
> 
> * Add this branch of PA to a new project with the LEGO Microgame, 2021.3.7f1
> * Open Project Auditor, go to Shader Variants window, click Clear
> * Build the game (I build OSX Standalone)
> * In the Shader Variants window, click Refresh.
> 
> DivideByZeroException: Attempted to divide by zero. Unity.ProjectAuditor.Editor.Modules.ShadersModule.AddShader 

Fixed.
